src/zenml/integrations/seldon/services/seldon_deployment.py

```diff
@@ -42,45 +44,107 @@ class SeldonDeploymentConfig(ServiceConfig):
+    # TODO [ENG-775]: have an enum of all supported Seldon Core implementations
     implementation: str
-    pipeline_name: Optional[str] = None
-    pipeline_run_id: Optional[str] = None
-    pipeline_step_name: Optional[str] = None
     replicas: int = 1
+    secret_name: Optional[str]
     model_metadata: Dict[str, Any] = Field(default_factory=dict)
     extra_args: Dict[str, Any] = Field(default_factory=dict)
 
-    # configuration attributes that are not part of the service configuration
-    # but are required for the service to function. These must be moved to the
-    # stack component, when available
-    kubernetes_context: Optional[str]
-    namespace: Optional[str]
-    base_url: str
-    # TODO [ENG-776]: replace with ZenML secret and create a k8s secret resource
-    #   that can be mounted in the container
-    secret_name: Optional[str]
+    def get_seldon_deployment_labels(self) -> Dict[str, str]:
+        """Generate the labels for the Seldon Core deployment from the
+        service configuration.
+
+        These labels are attached to the Seldon Core deployment resource
+        and may be used as label selectors in lookup operations.
+
+        Returns:
+            The labels for the Seldon Core deployment.
+        """
+        labels = {}
+        if self.pipeline_name:
+            labels["zenml.pipeline_name"] = self.pipeline_name
+        if self.pipeline_run_id:
+            labels["zenml.pipeline_run_id"] = self.pipeline_run_id
+        if self.pipeline_step_name:
+            labels["zenml.pipeline_step_name"] = self.pipeline_step_name
+        if self.model_name:
+            labels["zenml.model_name"] = self.model_name
+        if self.model_uri:
+            labels["zenml.model_uri"] = self.model_uri
+        if self.implementation:
+            labels["zenml.model_type"] = self.implementation
+        SeldonClient.sanitize_labels(labels)
+        return labels
+
+    def get_seldon_deployment_annotations(self) -> Dict[str, str]:
+        """Generate the annotations for the Seldon Core deployment from the
+        service configuration.
+
+        The annotations are used to store additional information about the
+        Seldon Core service that is associated with the deployment that is
+        not available in the labels. One annotation particularly important
+        is the serialized Service configuration itself, which is used to
+        recreate the service configuration from a remote Seldon deployment.
+
+        Returns:
+            The annotations for the Seldon Core deployment.
+        """
+        annotations = {
+            "zenml.service_config": self.json(),
+            "zenml.version": __version__,
+        }
+        return annotations
+
+    @classmethod
+    def create_from_deployment(
+        cls, deployment: SeldonDeployment
+    ) -> "SeldonDeploymentConfig":
+        """Recreate a Seldon Core service configuration from a Seldon Core
+        deployment resource.
```

Comment:
A Seldon Core deployment resource is a deployed seldon model? I feel with the sheer amount of nouns (deployment, service, configuration, seldon, core,  resource) makes my brain turn off.
```suggestion
        """Recreate the configuration of a Seldon Core Service from a deployed instance.
```

Would something like this work? It might be less accurate but easier to digest

Reply:
Yes, thank you. I tried to be as clear as I could, but it can be difficult to separate between all these names.